abr_analyze/nengo_utils/network_utils.py

```diff
@@ -15,490 +15,513 @@
+        arm (all joint position and velocities) and only returns the values
+        specified by the indices in in_index
+
+        EX: in_index = [0, 1, 3, 6]
+        will return the scaled joint position and velocities for joints 0,
+        1, 3 and 6
+    '''
+    #TODO: generalize this so the user passes in means and scales? right
+    #now this is specific for the jaco
+    #NOTE: do we want to have in_index here, or have the user specify what
+    # joint positions and velocities they pass in?
+    qs = q.T
+    dqs = dq.T
+
+    # expected mean of joint angles / velocities
+    means_q = np.array([0, 0, 0, 0, 0, 0])
+    means_dq = np.array([1.25, 1.25, 1.25, 1.25, 1.25, 1.25])
+
+    # expected variance of joint angles / velocities
+    scales_q = np.array([6.28, 6.28, 6.28, 6.28, 6.28, 6.28,])
+    scales_dq = np.array([2.5, 2.5, 2.5, 2.5, 2.5, 2.5])
+
+    for pp in range(0, 5):
+        qs[pp] = (qs[pp] - means_q[pp]) / scales_q[pp]
+        dqs[pp] = (dqs[pp] + means_dq[pp]) / scales_dq[pp]
+    qs = np.clip(qs, 0, 1)
+    dqs = np.clip(dqs, 0, 1)
+
+    scaled_q = np.array([qs[ii] for ii in in_index]).T
+    scaled_dq = np.array([dqs[ii] for ii in in_index]).T
+
+    return [scaled_q, scaled_dq]
+
+
+def raster_plot(network, input_signal, ax, n_ens_to_raster=None):
+    '''
+    Accepts a Nengo network and runs a simulation with the provided input_signal
+    Plots rasterplot onto ax object up to n_ens_to_raster ensembles
+    if n_ens_to_raster is None, all ensembles will be plotted
+
+    PARAMETERS
+    ----------
+    network: .DynamicsAdaptation
+        'abr_control.controllers.signals.dynamics_adaptation'
+    input_signal: np array shape of (time_steps x input_dim)
+        the input used for the network sim
+    ax: ax object
+        used for the rasterplot
+    n_ens_to_raster: int, Optional (Default: None)
+        the number of ensembles to plot in the raster, if None all will be plotted
+    '''
+    if n_ens_to_raster is None:
+        n_ens_to_raster = len(network.adapt_ens)
+
+    # create probes to get rasterplot
+    with network.nengo_model:
+        network.nengo_model.ens_probes = [
+            nengo.Probe(network.adapt_ens[ii].neurons, synapse=None)
+            for ii in range(n_ens_to_raster)]
+    sim = nengo.Simulator(network.nengo_model)
+
+    print('Running sim...')
+    for inputs in input_signal:
+        network.input_signal = inputs
+        sim.run(time_in_seconds=0.001, progress_bar=False)
+    print('Sim complete')
+
+    probes = []
+    print('Plotting spiking activity...')
+    for probe in network.nengo_model.ens_probes:
+        probes.append(sim.data[probe])
+
+    probes = np.hstack(probes)
+    time = np.ones(len(input_signal))
+    ax = rasterplot(np.cumsum(time), probes, ax=ax)
+    ax.set_ylabel('Neuron')
+    ax.set_xlabel('Time [sec]')
+    ax.set_title('Spiking Activity')
+
+
+def get_activities(network, input_signal):
```

Comment:
I thought this function needed to be changed to account for the neuron dynamics? This looks the same to me. Is it replaced with `get_spike_trains` and this function is just left for some other reason?

Reply:
correct! it's still useful for some things, i think. although if it's not used anywhere i can see removing it entirely